skbio/stats/distance/_regression.py

```diff
@@ -0,0 +1,170 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+import six
+import numpy as np
+from skbio.util._decorator import experimental
+from skbio.util._misc import check_random_state
+from skbio.stats.distance._mantel import _order_dms
+
+from skbio.stats.distance import DistanceMatrix
+
+@experimental(as_of="0.4.0")
+def linregress(y, *args, **kwargs):
+    """
+    This module performs a multiple linear regression on distance
+    matrices.  This is done by extracting the upper (or lower)
+    triangular portions of the symmetric distance matrices,
+    and collapsing them to compute a multiple linear regression
+    on the distances.  The signficances are computed by
+    permuting the residuals.
+
+    Parameters
+    ----------
+    y : DistanceMatrix, array_like, or filepath
+        Response distance matrix
+    x1, x2, ... : DistanceMatrix, array_like, or filepath
+        to distance matrices.  These distance matrices are predictors
+        for the regression.  They are also known as covariates
+    permutations : int, optional
+        Number of times to randomly permute `x` when assessing statistical
+        significance. Must be greater than or equal to zero. If zero,
+        statistical significance calculations will be skipped and the p-value
+        will be ``np.nan``.
+    random_state: int or np.RandomState, optional
+        Pseudo number generator state used for random sampling.
+    strict : bool, optional
+        If ``True``, raises a ``ValueError`` if IDs are found that do not exist
+        in all of the distance matrices. If ``False``, any nonmatching IDs are
+        discarded before running the test. See `n` (in Returns section below)
+        for the number of matching IDs that were used in the test. This
+        parameter is ignored if `x` and `y` are ``array_like``.
+    lookup : dict, optional
+        Maps each ID in the distance matrices to a new ID. Used to match up IDs
+        across distance matrices prior to running the Mantel test. If the IDs
+        already match between the distance matrices, this parameter is not
+        necessary. This parameter is disallowed if `x` and `y` are
+        ``array_like``.
+
+    Returns
+    -------
+    B : np.array
```

Comment:
That would be actually really awesome.

However, I'm not sure what the best way to do this is.  Each of the coefficients corresponds to a covariate distance matrix.  One option is to pass in a bunch of ids, one for each distance matrix.  But that would bulk up the number of input parameters and could make it confusing how to handle input parameters.

The other option is to pass in an optional ID into the DistanceMatrix constructor.  Any thoughts?